stripe/src/main/java/com/stripe/android/model/ShippingMethod.java

```diff
@@ -0,0 +1,110 @@
+package com.stripe.android.model;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.json.JSONObject;
+
+import java.util.Currency;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.stripe.android.model.StripeJsonUtils.putDoubleIfNotNull;
+import static com.stripe.android.model.StripeJsonUtils.putStringIfNotNull;
+
+/**
+ * Model representing a shipping method in the Android SDK
+ */
+public class ShippingMethod extends StripeJsonModel {
+
+    private static final String FIELD_AMOUNT = "amount";
+    /*ISO Currency Code*/
+    private static final String FIELD_CURRENCY_CODE = "currency_code";
+    private static final String FIELD_DETAIL = "detail";
+    private static final String FIELD_IDENTIFIER = "identifier";
+    private static final String FIELD_LABEL = "label";
+
+    private Double mAmount;
+    private String mCurrencyCode;
+    private @Nullable String mDetail;
+    private String mLabel;
+    private String mIdentifier;
+
+    public ShippingMethod(String label, String identifier, double amount, String currencyCode) {
+        this.mLabel = label;
+        this.mIdentifier = identifier;
+        this.mAmount = amount;
+        mCurrencyCode = currencyCode;
+    }
+
+    public ShippingMethod(String label,  String identifier, @Nullable String detail, double amount, String currencyCode) {
+        this.mLabel = label;
+        this.mIdentifier = identifier;
+        this.mDetail = detail;
+        this.mAmount = amount;
+        mCurrencyCode = currencyCode;
+    }
+
+    /**
+     * @return the currency that the specified amount will be rendered in.
+     */
+    public Currency getCurrency() {
+        return Currency.getInstance(mCurrencyCode);
+    }
+
+    /**
+     * @return The cost in major unit of the currency provided in the
+     * {@link com.stripe.android.PaymentConfiguration}. For example, dollars in the USA and yen in
+     * Japan.
+     */
+    public double getAmount() {
```

Comment:
Is there a reason you're going with a double for the currency amount, rather than a long in the lowest denomination like we usually do? ($1.00 -> 100L)